external/concealer/include/concealer/autoware_universe.hpp

```diff
@@ -0,0 +1,576 @@
+#include <boost/range/adaptor/sliced.hpp>
+#include <concealer/autoware.hpp>
+#include <concealer/conversion.hpp>
+#include <concealer/define_macro.hpp>
+#include <nav_msgs/msg/odometry.hpp>
+
+namespace concealer
+{
+class AutowareUniverse : public Autoware, public TransitionAssertion<AutowareUniverse>
+{
+  friend class TransitionAssertion<AutowareUniverse>;
+
+  void sendSIGINT() override;
+
+  bool isReady() noexcept;
+
+  bool isNotReady() noexcept;
+
+  void checkAutowareState();
+
+  bool is_ready = false;
+
+  /// FROM MiscellaneousAPI ///
+  //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+  /* ---- Checkpoint -----------------------------------------------------------
+   *
+   *  Set goal pose of Autoware.
+   *
+   *  Topic: /planning/mission_planning/checkpoint
+   *
+   * ------------------------------------------------------------------------ */
+  using Checkpoint = geometry_msgs::msg::PoseStamped;
+
+  DEFINE_PUBLISHER(Checkpoint);
+
+  /* ---- CurrentControlMode ---------------------------------------------------
+   *
+   *  Topic: /vehicle/status/control_mode
+   *
+   * ------------------------------------------------------------------------ */
+  using CurrentControlMode = autoware_auto_vehicle_msgs::msg::ControlModeReport;
+
+  DEFINE_PUBLISHER(CurrentControlMode);
+
+  decltype(auto) setCurrentControlMode(const std::uint8_t mode = CurrentControlMode::AUTONOMOUS)
+  {
+    CurrentControlMode current_control_mode{};
+    {
+      current_control_mode.mode = mode;
+    }
+
+    return setCurrentControlMode(current_control_mode);
+  }
+
+  /* ---- CurrentShift ---------------------------------------------------------
+   *
+   *  Topic: /vehicle/status/gear_status
+   *
+   *  Overloads:
+   *    setCurrentShift(const autoware_auto_vehicle_msgs::msg::GearReport &);
+   *    setCurrentShift(const double);
+   *
+   * ------------------------------------------------------------------------ */
+  using CurrentShift = autoware_auto_vehicle_msgs::msg::GearReport;
+
+  DEFINE_PUBLISHER(CurrentShift);
+
+  template <typename T, REQUIRES(std::is_floating_point<T>)>
+  decltype(auto) setCurrentShift(const T twist_linear_x)
+  {
+    CurrentShift current_shift;
+    {
+      using autoware_auto_vehicle_msgs::msg::GearReport;
+
+      current_shift.stamp = get_clock()->now();
+      current_shift.report = twist_linear_x >= 0 ? GearReport::DRIVE : GearReport::REVERSE;
+    }
+
+    return setCurrentShift(current_shift);
+  }
+
+  decltype(auto) setCurrentShift(const geometry_msgs::msg::Twist & twist)
+  {
+    return setCurrentShift(twist.linear.x);
+  }
+
+  /* ---- CurrentSteering ------------------------------------------------------
+   *
+   *  Topic: /vehicle/status/steering_status
+   *
+   * ------------------------------------------------------------------------ */
+  // TODO(murooka) steering is tire angle, not tire angular velocity
```

Comment:
I'm not sure why this function used not tire angle but tire angular velocity.
We have to modify this code to get tire angle (from directly ackermann control command?)

Reply:
Maybe this topic is just used to move forward to WaitingForRoute from InitializingVehicle.
Just in case, I modified the code so that psim_v2 publishes steering tire angle which is the same as commanded one from Autoware.
https://github.com/tier4/scenario_simulator_v2/pull/617/commits/444ff51c857e0a941905dfc1451d63a92eb4c7a2